include/renderers/gl/mir/renderer/gl/texture_target.h

```diff
@@ -34,6 +34,9 @@ class TextureTarget
     /* Bind the target as a texture suitable for rendering to via glFramebufferTexture2D */
     virtual void bind_for_write() = 0;
 
+    /// Memory backed targets need to secure the pixels after rendering
+    virtual void secure_pixels() = 0;
```

Comment:
My only other thought is secure_for_read(), as I expect the typical use is to read() right after.